packages/core/src/modules/proofs/protocol/v2/messages/V2RequestPresentationMessage.ts

```diff
@@ -0,0 +1,118 @@
+import type { ProofAttachmentFormat } from '../../../formats/models/ProofAttachmentFormat'
+
+import { Expose, Type } from 'class-transformer'
+import { IsArray, IsBoolean, IsInstance, IsOptional, IsString, ValidateNested } from 'class-validator'
+
+import { AgentMessage } from '../../../../../agent/AgentMessage'
+import { Attachment } from '../../../../../decorators/attachment/Attachment'
+import { AriesFrameworkError } from '../../../../../error'
+import { IsValidMessageType, parseMessageType } from '../../../../../utils/messageType'
+import { uuid } from '../../../../../utils/uuid'
+import { ProofFormatSpec } from '../../../formats/models/ProofFormatSpec'
+
+export interface V2RequestPresentationMessageOptions {
+  id?: string
+  comment?: string
+  goalCode?: string
+  presentMultiple?: boolean
+  willConfirm?: boolean
+  attachmentInfo: ProofAttachmentFormat[]
+}
+
+export class V2RequestPresentationMessage extends AgentMessage {
+  public constructor(options: V2RequestPresentationMessageOptions) {
+    super()
+
+    this.formats = []
+    this.requestPresentationsAttach = []
```

Comment:
I think we should only set these if options is defined (this is for all messages classes)